pkg/providers/vsphere/virtualmachine/backup.go

```diff
@@ -53,22 +56,40 @@ func BackupVirtualMachine(opts BackupVirtualMachineOptions) error {
 		opts.VMCtx.Logger.Error(err, "failed to get VM properties for backup")
 		return err
 	}
-
 	curExCfg := pkgutil.OptionValues(moVM.Config.ExtraConfig)
 	var ecToUpdate pkgutil.OptionValues
 
-	vmYAML, err := getDesiredVMResourceYAMLForBackup(opts.VMCtx.VM, curExCfg)
-	if err != nil {
-		opts.VMCtx.Logger.Error(err, "failed to get VM resource yaml for backup")
-		return err
-	}
-	if vmYAML == "" {
-		opts.VMCtx.Logger.V(4).Info("Skipping VM resource yaml backup as unchanged")
-	} else {
-		ecToUpdate = append(ecToUpdate, &vimtypes.OptionValue{
-			Key:   vmopv1.VMResourceYAMLExtraConfigKey,
-			Value: vmYAML,
-		})
+	/*
+	 * When VMIncrementalRestore FSS is enabled, perform backups when
+	 * - the backup version annotation matches the current backup version in extraConfig.
+	 * - no backup version annotation is present.
+	 * - the backup version annotation is older than the extraConfig backup version key. This can only happen when reconfigure succeeds to add a new
+	 *   extraConfig backup version but the next reconcile failed to update the annotation on the VM resource.
+	 */
+	if pkgcfg.FromContext(opts.VMCtx).Features.VMIncrementalRestore {
+		if backupVersionAnnotation, ok := opts.VMCtx.VM.Annotations[vmopv1.VirtualMachineBackupVersionAnnotation]; ok && backupVersionAnnotation != "" {
+			currBackupVersionEcKey, _ := curExCfg.GetString(vmopv1.BackupVersionKey)
```

Comment:
Suggest use `backupVersionEcKey` instead of `currBackupVersionEcKey` since there is only one `backupVersionEcKey` and one `backupVersionAnnotation`. It can be confused with later `currVersion`.

Reply:
Also, the var should be `backupVersionEcVal` instead of Key?

Reply:
> Should you also check for the case where the EC doesn't exist (the second return value of the GetString method)?

@aruneshpa , Do we care if the EC is doesn't exist vs ""?. Either ways at the moment the implementation decides to do a backup and fix it.